### ISSUE TITLE
Getting Invalid use of BasicClientConnManager: in parallel environment

### ISSUE THREAD

**Issue by kushalpal17** [dzieciou/curl-logger#37]
```
mvn gauge:execute -DspecsDir="specs/" -DinParallel=true -Dnodes=4


java.lang.IllegalStateException: Invalid use of BasicClientConnManager: connection still allocated.
Make sure to release the connection before allocating another one.
```

**Comment by dzieciou** [comment]
Thanks for reporting the problem.

1. Does the problem occur if you don't use curl logger? If so, it might be related to how RestAssured uses Apache HTTP Client.
2. If no, can you provide me minimal working example so I can reproduce the case? Especially what's your Maven configuration (`pom.xml`) and what's inside of specs/?

**Comment by dzieciou** [comment]
I was able to reproduce the bug following instructions @francislainy . This seems to be a bug in REST-assured that I reported there: https://github.com/rest-assured/rest-assured/issues/1392. REST-assured is not releasing HTTP connections when reusing HTTP client instance.

A workaround would be to remove `reuseHttpClientInstance()` in `CurlRestAssuredConfigFactory`, but I would prefer this to be fixed in REST-assured. There are reasons to reuse HTTP client instance across multiple requests, namely, performance reasons. You don't want to introduce additional overhead.
